### ISSUE TITLE
Colon separated class bindings in Vue does not generate the expected class

### ISSUE THREAD

**Issue by Oskar-Nilsen-Roos** [tailwindlabs/tailwindcss#13492]
**What version of Tailwind CSS are you using?**

v3.4.3

**What build tool (or framework if it abstracts the build tool) are you using?**

--

**What version of Node.js are you using?**

--

**What browser are you using?**

N/A

**What operating system are you using?**

macOS

**Reproduction URL**

https://play.tailwindcss.com/JBhjflrJ2x

**Describe your issue**

Using a [Vue class binding](https://vuejs.org/guide/essentials/class-and-style#binding-to-objects) like so:
```HTML
<div :class="{ invisible: hideClose }">
</div>
```

Will result in the following minified Javascript when built:
```Javascript
{class:l({invisible:e.hideClose})}
```

Any class not surrounded by double or single quotes, and separated by a colon on either side with a following value will not generate the expected tailwind class.

Expected result:
Generated 'invisible' class.

Actual result:
No 'invisible' class generated.

Simply surrounding the value in quotes would fix it in this case, however building the file will minify it away since it's not technically necessary.

Only current workaround I know of is not using a Vue class binding, and instead using a ternary operator or something similar, leaving you with quotes in the minified file:
```HTML
<div :class="[hideClose ? 'invisible' : '']">
</div>
```

Giving you:
```Javascript
{class:l([e.hideClose?"invisible":""])}
```

Thanks!

**Comment by adamwathan** [comment]
Hey! What's the situation that's causing you to run Tailwind against minified JS instead of against the original source files?

**Comment by Oskar-Nilsen-Roos** [comment]
> Hey! What's the situation that's causing you to run Tailwind against minified JS instead of against the original source files?

Right, should've clarified! In our use-case we have a Vue components library which is tree-shaken and then consumed by 10+ applications which make up individual fragments of a website (usually called micro frontends). We then generate the css files for each application as it's built, by pointing the `contents` prop in the tailwind config to the built js file of the application.

So one solution would definitely be to go against the source files, generate a `components.css` and then include that css with each application. However, this would cause a lot of css duplication on every single page.

I understand that this is a niche issue, and probably not a way you intended for it to be used, so I don't expect you to change the code to suite this need!

Can you see any other way to solve this while avoiding as much code duplication as possible? Or maybe come across someone solving using tailwind in a similar structure?

Thank you!

**Comment by philipp-spiess** [comment]
@Oskar-Nilsen-Roos I think in your setup it would be best if you do not minify the Vue component library, I do think you could probably still apply tree-shaking if you want to but I imagine that each of your 10+ application already runs a minifier as part of the production build anyways.

Unfortunately operating on minified sources is really not something we're looking to add to support with for now since it would mean we have to remove a lot of heuristics we apply to improve performance dramatically for everyone. :/

Going to close this issue since it's not something we intend to change for now but feel free to open a discussion if you want more help. Also really sorry that this issue slipped through the cracks and you haven't gotten a reply for more than 10 months.
